server/src/main/java/org/opensearch/index/store/FsDirectoryFactory.java

```diff
@@ -97,9 +97,10 @@ protected Directory newFSDirectory(Path location, LockFactory lockFactory, Index
             case HYBRIDFS:
                 // Use Lucene defaults
                 final FSDirectory primaryDirectory = FSDirectory.open(location, lockFactory);
+                final Set<String> mmapExtensions = new HashSet<>(indexSettings.getValue(IndexModule.INDEX_STORE_HYBRID_MMAP_EXTENSIONS));
```

Comment:
@mattweber I am trying to wrap my head around these two properties `INDEX_STORE_PRE_LOAD_SETTING` and `INDEX_STORE_HYBRID_MMAP_EXTENSIONS`, which apparently serve the same purpose (and do have same set of values), the `HybridDirectory` is looking like `PreLoadMMapDirectory` (except the `setPreload` thing) ...

Reply:
@reta  `INDEX_STORE_PRE_LOAD_SETTING` controls which files automatically get loaded into os cache and I don't believe this has any defaults as it is an expensive operation.  This PR adds `INDEX_STORE_HYBRID_MMAP_EXTENSIONS` which controls which extensions are with opened with mmap store.  The current defaults were picked by elastic after doing benchmarks.  I would like it configurable so a user can tune to their specific usecase.  For example, I want to mmap everything EXCEPT stored fields.  The settings are related, but not the same.